### PR TITLE
createMetadataContainerIfNotExistsForMasterKeyAuth

### PR DESCRIPTION
**Pull request by xinlian12**
Adding support for creating metadata container if not exists when using masterKey auth, for other auth types (AAD), it will throw an exception.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-kafka-connect/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #### Bugs Fixed
 
 #### Other Changes
+* Added support to create metadata container if not exists for MasterKey auth - See [PR 39973](https://github.com/Azure/azure-sdk-for-java/pull/39973)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-kafka-connect/CHANGELOG.md:12]: We should clarify metadata container for which feature (throughput control in this case).

File: sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSourceConnector.java [hunk 2]
```diff
@@ -167,13 +171,37 @@ private IMetadataReader getMetadataReader() {
...
                     .block();
                 return new MetadataCosmosStorageManager(metadataContainer);
             default:
                 throw new IllegalArgumentException("Metadata storage type " + this.config.getMetadataConfig().getStorageType() + " is not supported");
         }
     }
 
+    private boolean shouldCreateMetadataContainerIfNotExists() {
+        // If customer does not create the metadata container ahead of time,
+        // then SDK will create one with default autoScale config only if using masterKey auth.
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSourceConnector.java:191]: I recall Fabric team mentioning to go ahead with default 400 RUs, no? I know we have seen problems with throttling with 400 RUs, but curious are they blockers?

File: sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSourceConnector.java [hunk 3]
```diff
@@ -167,13 +171,37 @@ private IMetadataReader getMetadataReader() {
...
+            && (this.config.getAccountConfig().getCosmosAuthConfig() instanceof CosmosMasterKeyAuthConfig);
+    }
+
+    private Mono<CosmosContainerResponse> createMetadataContainer() {
+        return this.cosmosClient
+            .getDatabase(this.config.getContainersConfig().getDatabaseName())
+            .createContainer(
+                this.config.getMetadataConfig().getStorageName(),
+                "/id",
+                ThroughputProperties.createAutoscaledThroughput(4000));
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSourceConnector.java:202]: Better to put this RU in a constant variable and reference it here.

**Review by kushagraThapar — APPROVED** [review]
LGTM with few suggestions, thanks @xinlian12

File: sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSourceConnector.java (hunk 2 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSourceConnector.java:191]: they proposed using 400RUs, but we decided to use autoscale 4000RUs

File: sdk/cosmos/azure-cosmos-kafka-connect/CHANGELOG.md (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-kafka-connect/CHANGELOG.md:12]: the metadata is not related to throughput , the metadata container is used to track metadata (like the containerRids, feedRanges for the container),

**Pull request merged**